livecheck/special/maven.py

```diff
@@ -0,0 +1,49 @@
+"""Maven functions."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+import logging
+import subprocess as sp
+
+from livecheck.utils import check_program
+
+from .utils import build_compress, remove_url_ebuild, search_ebuild
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+
+__all__ = ('check_maven_requirements', 'remove_maven_url', 'update_maven_ebuild')
+
+log = logging.getLogger(__name__)
+
+
+def remove_maven_url(ebuild_content: str) -> str:
+    """Remove the URL for the Maven dependency tarball from the ebuild content."""
+    return remove_url_ebuild(ebuild_content, '-mvn.tar.xz')
+
+
+def update_maven_ebuild(ebuild: str, path: str | None, fetchlist: Mapping[str, tuple[str,
+                                                                                     ...]]) -> None:
+    """Update a Maven package ebuild."""
+    maven_path, temp_dir = search_ebuild(ebuild, 'pom.xml', path)
+    if not maven_path:
+        return
+
+    try:
+        sp.run(('mvn', '--batch-mode', '-Dmaven.repo.local=.m2', 'dependency:go-offline',
```

Comment:
Do we maybe want a better argument to `-Dmaven.repo.local`? Something like: `platformdirs.user_cache_path() / 'livecheck/.m2'`?

Reply:
No, because the working directory of the ebuild must be relative, otherwise you cannot generate the tar